### PR TITLE
Add 2023 Maryland Tax Rules

### LINKED ISSUE #4181: Add 2023 Maryland Tax Rules
2023 Income Tax Form Instructions: https://www.marylandtaxes.gov/forms/23_forms/Resident-Booklet.pdf
(2023 Maryland Child Tax Credit has been already been added)

### PR DESCRIPTION
**Pull request by Tianbk**
Fixes #4181

### CODE REVIEW

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:1]: add a changelog_entry file

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml [hunk 1]
```diff
@@ -18,15 +20,20 @@ metadata:
 JOINT:
   2018-01-01: 4_500
   2021-01-01: 4_700
+  2023-01-01: 5_150
 HEAD_OF_HOUSEHOLD:
   2018-01-01: 4_500
   2021-01-01: 4_700
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml:26]: check 2022 file for the 2022 vlaues:
https://marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf

Also add uprating metadata:
example:
  uprating:
    parameter: gov.irs.uprating
    rounding:
      type: downwards
      interval: 50

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml:26]: same for the min file

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml [hunk 2]
```diff
@@ -1,8 +1,11 @@
 description: MD maximum pension AGI subtraction
 values:
   2021-01-01: 34_300
+  2023-01-01: 36_200
 metadata:
   reference:
+    - title: "PENSION EXCLUSION COMPUTATION WORKSHEET(13A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml:7]: could you check the 2022 tax form to see if we missed a 2022 adjustment?

File: policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/disabled.yaml [hunk 3]
```diff
@@ -6,5 +6,7 @@ metadata:
   unit: year
   period: year
   reference:
+    - title: "8. MARYLAND CHILD TAX CREDIT."
+      href: https://www.marylandtaxes.gov/forms/23_forms/Resident-Booklet.pdf#page=20
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/disabled.yaml:10]:
```suggestion
      href: https://www.marylandtaxes.gov/forms/23_forms/Resident-Booklet.pdf#page=25
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/disabled.yaml:10]: Please double check all pages - this one randomly picked was off - need to use the page where the parameter appears

File: policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/disabled.yaml [hunk 4]
```diff
@@ -6,5 +6,7 @@ metadata:
   unit: year
   period: year
   reference:
+    - title: "8. MARYLAND CHILD TAX CREDIT."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/disabled.yaml:9]:
```suggestion
    - title: Maryland 2023 Resident Tax Forms and Instructions 
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/disabled.yaml:9]: use the title of the actual form in all titles of the references

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml [hunk 5]
```diff
@@ -3,9 +3,12 @@ metadata:
   label: Maryland maximum standard deduction
   period: year
   unit: currency-USD
+  uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml:6]: the uprating should be applied to each individual filing status - all breakdown parameters

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml [hunk 6]
```diff
@@ -6,6 +6,8 @@ metadata:
   breakdown:
     - filing_status
   reference:
+    - title: "Standard Deduction Worksheet (16A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml:9]:
```suggestion
    - title: Maryland 2023 State & Local Tax Forms & Instructions
```

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml [hunk 7]
```diff
@@ -3,9 +3,12 @@ metadata:
   label: Maryland minimum standard deduction
   period: year
   unit: currency-USD
+  uprating: gov.irs.uprating
   breakdown:
     - filing_status
   reference:
+    - title: "Standard Deduction Worksheet (16A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml:10]:
```suggestion
    - title: Maryland 2023 State & Local Tax Forms & Instructions
```

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml [hunk 8]
```diff
@@ -3,9 +3,12 @@ metadata:
   label: Maryland minimum standard deduction
   period: year
   unit: currency-USD
+  uprating: gov.irs.uprating
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml:6]: this looks like it should be rounded

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml (hunk 5 above)
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/max.yaml:6]:
```suggestion
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/max_two_income_subtraction.yaml [hunk 9]
```diff
@@ -3,6 +3,8 @@ values:
   2021-01-01: 1_200
 metadata:
   reference:
+    - title: "Line 14. TWO-INCOME SUBTRACTION."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/max_two_income_subtraction.yaml:6]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml [hunk 10]
```diff
@@ -1,9 +1,14 @@
 description: MD maximum pension AGI subtraction
 values:
   2021-01-01: 34_300
+  2023-01-01: 36_200
 metadata:
   reference:
-    - title: "Form 502, line 10a"
+    - title: "2023 PENSION EXCLUSION COMPUTATION WORKSHEET(13A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml:7]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml [hunk 11]
```diff
@@ -1,9 +1,14 @@
...
+      href: https://www.marylandtaxes.gov/forms/23_forms/Resident-Booklet.pdf#page=13
+    - title: "2022 PENSION EXCLUSION COMPUTATION WORKSHEET(13A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml:9]:
```suggestion
    - title: 2022 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml [hunk 12]
```diff
@@ -1,9 +1,14 @@
...
   2021-01-01: 34_300
+  2023-01-01: 36_200
 metadata:
   reference:
-    - title: "Form 502, line 10a"
+    - title: "2023 PENSION EXCLUSION COMPUTATION WORKSHEET(13A)"
+      href: https://www.marylandtaxes.gov/forms/23_forms/Resident-Booklet.pdf#page=13
+    - title: "2022 PENSION EXCLUSION COMPUTATION WORKSHEET(13A)" 
+      href: https://www.marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf#page=12
+    - title: "2021 PENSION EXCLUSION COMPUTATION WORKSHEET(13A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/max_amount.yaml:11]:
```suggestion
    - title: 2021 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/min_age.yaml [hunk 13]
```diff
@@ -3,6 +3,8 @@ values:
   2021-01-01: 65
 metadata:
   reference:
+    - title: "Line 10a. PENSION EXCLUSION."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/pension/min_age.yaml:6]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/main.yaml [hunk 14]
```diff
@@ -7,5 +7,7 @@ metadata:
   unit: year
   period: year
   reference:
+    - title: "8. MARYLAND CHILD TAX CREDIT."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/age_threshold/main.yaml:10]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/amount.yaml [hunk 15]
```diff
@@ -6,5 +6,7 @@ metadata:
   unit: currency-USD
   period: year
   reference:
+    - title: "8. MARYLAND CHILD TAX CREDIT."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/ctc/amount.yaml:9]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/married_or_has_child/match.yaml [hunk 16]
```diff
@@ -5,6 +5,8 @@ metadata:
   unit: /1
   label: Maryland non-refundable EITC match
   reference:
+    - title: "STATE EARNED INCOME CREDIT WORKSHEET (18A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/non_refundable/married_or_has_child/match.yaml:8]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/married_or_has_child/match.yaml [hunk 17]
```diff
@@ -10,5 +10,7 @@ metadata:
   unit: /1
   label: Maryland refundable EITC match
   reference:
+    - title: "Line 42. REFUNDABLE EARNED INCOME CREDIT."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/eitc/refundable/married_or_has_child/match.yaml:13]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/poverty_line/earned_income_share.yaml [hunk 18]
```diff
@@ -5,5 +5,7 @@ metadata:
   unit: /1
   label: Maryland Poverty Line Credit rate
   reference:
+    - title: "Line 23. STATE POVERTY LEVEL CREDIT."
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/poverty_line/earned_income_share.yaml:8]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/senior_tax/age_eligibility.yaml [hunk 19]
```diff
@@ -3,6 +3,8 @@ values:
   2022-01-01: 65
 metadata:
   reference:
+    - title: MD 2023 Resident Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/senior_tax/age_eligibility.yaml:6]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/senior_tax/amount/head_of_household.yaml [hunk 20]
```diff
@@ -3,6 +3,8 @@ values:
   2022-01-01: 1_750
 metadata:
   reference:
+    - title: MD 2023 Resident Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/senior_tax/amount/head_of_household.yaml:6]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/credits/senior_tax/amount/joint.yaml [hunk 21]
```diff
@@ -1,6 +1,8 @@
 description: Maryland provides this senior tax credit amount for joint filers, based on the number of aged people.
 metadata:
   reference:
+    - title: MD 2023 Resident Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/credits/senior_tax/amount/joint.yaml:4]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/exemptions/personal/separate.yaml [hunk 22]
```diff
@@ -35,5 +35,7 @@ metadata:
   amount_unit: currency-USD
   label: Maryland personal exemption for married filing separate filers, based on AGI
   reference:
+    - title: "EXEMPTION AMOUNT CHART (10A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/exemptions/personal/separate.yaml:38]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

File: policyengine_us/parameters/gov/states/md/tax/income/exemptions/personal/single.yaml [hunk 23]
```diff
@@ -35,5 +35,7 @@ metadata:
   amount_unit: currency-USD
   label: Maryland personal exemption for single filers, based on AGI
   reference:
+    - title: "EXEMPTION AMOUNT CHART (10A)"
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/exemptions/personal/single.yaml:38]:
```suggestion
    - title: 2023 State & Local Tax Forms & Instructions Maryland
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]
Please make sure this is appleid everywhere where relevant

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: `git pull upstream master`

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml [hunk 24]
```diff
@@ -16,17 +18,52 @@ metadata:
...
-  2021-01-01: 1_550
+  values:
+    2018-01-01: 1_500
+    2021-01-01: 1_550
+    2023-01-01: 1_700
+  metadata:
+    uprating: gov.irs.uprating
+    rounding:
+      type: downwards
+      interval: 50
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml:59]: can we nest the uprating metadata like this everywhere in the PR?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/sources.yaml [hunk 25]
```diff
@@ -1,4 +1,4 @@
-description: Maryland list of AGI subtractions.
+description: Maryland subtarcts the following sources from federal adjusted gross income.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/sources.yaml:1]:
```suggestion
description: Maryland subtracts the following sources from federal adjusted gross income.
```

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml [hunk 26]
```diff
@@ -16,17 +18,57 @@ metadata:
       href: https://www.marylandtaxes.gov/new-tax-year-update.php#:~:text=Standard%20Deduction%20%2D%20The%20tax%20year,spouse%2C%20and%20taxpayers%20filing%20jointly.
 
 JOINT:
-  2018-01-01: 3_000
-  2021-01-01: 3_100
+  values:
+    2018-01-01: 3_000
+    2021-01-01: 3_100
+    2023-01-01: 3_450
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml:24]: didn't change in 2022? please add a 2022 tax form ref

File: changelog_entry.yaml [hunk 27]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+      - Maryland 2023 income Tax updates.
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
      - Maryland 2023 income tax updates.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml (hunk 26 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/md/tax/income/deductions/standard/min.yaml:27]: Missed that, thank you, fixed

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**